src/Utilities/UnitTests/CommandLineBuilder_Tests.cs

```diff
@@ -405,19 +405,19 @@ public void AppendSwitchUnQuotedTaskItem()
         }
 
         /// <summary>
-        /// Odd number of literal quotes. This should trigger an exception, because command line parsers
-        /// generally can't handle this case.
+        /// Ensure it's not an error to have an odd number of literal quotes. Sometimes
+        /// it's a mistake on the programmer's side, but we cannot reject odd numbers of
+        /// quotes in the general case because sometimes that's exactly what's needed (e.g.
+        /// passing a string with a single embedded double-quote to a compiler).
         /// </summary>
         [Fact]
         public void AppendSwitchWithOddNumberOfLiteralQuotesInParameter()
         {
-            Assert.Throws<ArgumentException>(() =>
-            {
-                CommandLineBuilder c = new CommandLineBuilder();
-                c.AppendSwitchIfNotNull("/D", @"ASSEMBLY_KEY_FILE=""c:\\foo\\FinalKeyFile.snk");
-            }
-           );
+            CommandLineBuilder c = new CommandLineBuilder();
+            c.AppendSwitchIfNotNull("/D", @"A='""'");
+            Assert.Equal(@"/D""A='\""'""", c.ToString());
```

Comment:
I started typing this comment to say "use regular string literals" but halfway through translating them I decided they weren't any clearer.

Maybe it's worth adding the no-escape strings as a comment, though? It's definitely character soup in there.